### ISSUE TITLE
TransactionResponse.Ledger should be of type int instead of long

### ISSUE THREAD

**Issue by orbitlens** [elucidsoft/dotnet-stellar-sdk#183]
**Describe the bug**
Ledger sequence numbers are encoded as `Int32` internally in all XDR structures. However, the property `TransactionResponse.Ledger` is defined as `long`.

**Comment by Kirbyrawr** [comment]
Working on this, just to let you know.

**Comment by fracek** [comment]
From the XDR definition it looks like ledger sequence numbers are encoded as `uint32`, which is an `unsigned int`. I checked and we convert the `long` to `unsigned int` before serializing, which is fine as long as the sequence number is nonnegative.

**Comment by elucidsoft** [comment]
Right, which is why I don't think we should make a code change for this?

**Comment by elucidsoft** [comment]
So after looking into this I believe the XDR structure we have is incorrect. The XDR should be long as in the original XDR from Stellar-ledger-entries.x its an int64 which would be a long.  Am Imissing something?

https://github.com/stellar/stellar-core/blob/236f831521b6724c0ae63906416faa997ef27e19/src/xdr/Stellar-ledger-entries.x

**Comment by michaeljmonte** [comment]
```
struct LedgerHeader
{
    uint32 ledgerVersion;    // the protocol version of the ledger
    Hash previousLedgerHash; // hash of the previous ledger header
    StellarValue scpValue;   // what consensus agreed to
    Hash txSetResultHash;    // the TransactionResultSet that led to this ledger
    Hash bucketListHash;     // hash of the ledger state

    uint32 ledgerSeq; // sequence number of this ledger

    int64 totalCoins; // total number of stroops in existence.
                      // 10,000,000 stroops in 1 XLM

    int64 feePool;       // fees burned since last inflation run
    uint32 inflationSeq; // inflation sequence number

    uint64 idPool; // last used global ID, used for generating objects

    uint32 baseFee;     // base fee per operation in stroops
    uint32 baseReserve; // account base reserve in stroops

    uint32 maxTxSetSize; // maximum size a transaction set can be

    Hash skipList[4]; // hashes of ledgers in the past. allows you to jump back
                      // in time without walking the chain back ledger by ledger
                      // each slot contains the oldest ledger that is mod of
                      // either 50  5000  50000 or 500000 depending on index
                      // skipList[0] mod(50), skipList[1] mod(5000), etc

    // reserved for future use
    union switch (int v)
    {
    case 0:
        void;
    }
    ext;
};

Maybe I am missing something because here the ledger seq number looks like an Int.

**Comment by michaeljmonte** [comment]
This honestly looks like an oversight on the part of the Stellar/Horizon team.

**Comment by fracek** [comment]
I think you are confusing account sequence numbers (`int64`) with ledger sequence number (`uint32`). We use long for ledger sequence numbers, which is fine (the java sdk does the same).

**Comment by elucidsoft** [comment]
I think what we have is correct, there are language disparities that are causing confusion.  I'll ping @orbitlens on Keybase to discuss further.

**Comment by orbitlens** [comment]
Ledger sequence numbers are represented and stored on the ledger in a 4-bytes format. Many things internally rely on this value to be no more than UInt32.MaxValue. For example, Horizon transaction/operation/effect IDs and paging tokens generation algorithm expects strictly 4 bytes.
 
It seems to me that representing this value as `UInt32` or `Int32` is more consistent than treating it as `Int64`. Currently, we have to use explicit typecasting across all `/ledger` and `/transaction` endpoint responses.

**Comment by elucidsoft** [comment]
I would be more willing to do uint than int.

**Comment by fracek** [comment]
It should be `uint32` because `Int32.MaxValue < Uint32.MaxValue`.

**Comment by elucidsoft** [comment]
Correct.

**Comment by elucidsoft** [comment]
This is still up for grabs for whoever wants to take it!

**Comment by Kirbyrawr** [comment]
I will take it after work @elucidsoft

**Comment by fracek** [comment]
Was fixed in #189
